File: solutions/python/transpose/1/transpose.py
# With deepest reverence to Shree DR.MDD

def transpose(lines: str) -> str:
    block: list[str] = []
    max_len = 0
    for text in reversed(lines.splitlines()):
        max_len = max(len(text), max_len)
        block.append(text.ljust(max_len))
    block.reverse()

    grid: list[list[str]] = []
    for seq in block:
        grid.append(list(seq) + [""] * (max_len - len(seq)))

    return "\n".join(
        "".join(grid[row][col] for row in range(len(grid)))
        for col in range(max_len)
    )

File: solutions/javascript/transpose/1/transpose.js
// For my Shree DR.MDD

export const transpose = (lines = []) => {
  if (lines.length === 0) return [];

  let maxLen = 0;
  for (let idx = 0; idx < lines.length; idx++) {
    maxLen = Math.max(maxLen, lines[idx].length);
  }

  for (let idx = 0; idx < lines.length; idx++) {
    lines[idx] = lines[idx].replace(/ /g, '_');
  }

  for (let idx = 0; idx < lines.length; idx++) {
    lines[idx] = lines[idx].padEnd(maxLen, ' ');
  }

  let transposed = [];
  for (let i = 0; i < maxLen; i++) {
    let newRow = '';
    for (let j = 0; j < lines.length; j++) {
      newRow += lines[j][i];
    }
    transposed.push(newRow);
  }

  return transposed.map(row => row.trimEnd()).map(row => row.replaceAll("_", ' '));
};
